tests/e2e/common/diagnosticcollector.go

```diff
@@ -0,0 +1,465 @@
+		if _, wErr := fmt.Fprintf(c.errLog, "Error writing data: %v\n", err); wErr != nil {
+			return errors.Join(err, wErr)
+		}
+	}
+
+	return nil
+}
+
+func (c nodes) Collect(ctx context.Context) error {
+	nodes, err := c.clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		if _, wErr := fmt.Fprintf(c.errLog, "Error getting nodes: %v\n", err); wErr != nil {
+			return errors.Join(err, wErr)
+		}
+	}
+
+	nodesFile, err := c.createFile("nodes.yaml")
+	if err != nil {
+		return err
+	}
+	defer nodesFile.Close()
+
+	if err = c.writeToFile(nodesFile, nodes); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c namespaces) Collect(ctx context.Context) error {
+	namespaces, err := c.Config.clientset.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		if _, wErr := fmt.Fprintf(c.errLog, "Error getting namespaces: %v\n", err); wErr != nil {
+			return errors.Join(err, wErr)
+		}
+	}
+
+	namespacesFile, err := c.createFile("namespaces.yaml")
+	if err != nil {
+		return err
+	}
+	defer namespacesFile.Close()
+
+	if err = c.writeToFile(namespacesFile, namespaces); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c daemonsets) Collect(ctx context.Context) error {
+	daemonsets, err := c.clientset.AppsV1().DaemonSets(c.namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		if _, wErr := fmt.Fprintf(c.errLog, "Error getting daemonsets: %v\n", err); wErr != nil {
+			return errors.Join(err, wErr)
+		}
+	}
+
+	daemonsetsFile, err := c.createFile("daemonsets.yaml")
+	if err != nil {
+		return err
+	}
+	defer daemonsetsFile.Close()
+
+	if err = c.writeToFile(daemonsetsFile, daemonsets); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c deployments) Collect(ctx context.Context) error {
+	deployments, err := c.clientset.AppsV1().Deployments(c.namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		if _, wErr := fmt.Fprintf(c.errLog, "Error getting deployments: %v\n", err); wErr != nil {
+			return errors.Join(err, wErr)
+		}
+	}
+
+	deploymentsFile, err := c.createFile("deployments.yaml")
+	if err != nil {
+		return err
+	}
+	defer deploymentsFile.Close()
+
+	if err = c.writeToFile(deploymentsFile, deployments); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c pods) Collect(ctx context.Context) error {
+	pods, err := c.Config.clientset.CoreV1().Pods(c.namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		if _, wErr := fmt.Fprintf(c.errLog, "Error getting pods: %v\n", err); wErr != nil {
+			return errors.Join(err, wErr)
+		}
+	}
+
+	podsFile, err := c.createFile("pods.yaml")
+	if err != nil {
+		return err
+	}
+	defer podsFile.Close()
+
+	if err = c.writeToFile(podsFile, pods); err != nil {
+		return err
+	}
+
+	// Get all logs per pod
+	for _, pod := range pods.Items {
+		podLogFile, err := c.createFile(fmt.Sprintf("%s_logs.log", pod.Name))
+		if err != nil {
+			return err
+		}
+		defer podLogFile.Close()
+
+		podLogOpts := v1.PodLogOptions{}
```

Comment:
Any reason for this local?

Reply:
we are not setting any opts, so is to have the same Opts for each call on the loop when running `req := c.clientset.CoreV1().Pods(c.namespace).GetLogs(pod.Name, &podLogOpts)`

Reply:
OK. I don't really follow, because this isntantiation is IN the loop. Does this not mean that there is no difference between what we have here and:
```
req := c.clientset.CoreV1().Pods(c.namespace).GetLogs(pod.Name, &v1.PodLogOptions{})
```